src/util/fields.py

```diff
@@ -153,6 +153,31 @@ def clean_context(context):
         if changed:
             add_to_migration_reports(("ir.filters", id_, name), "Filters/Dashboards")
 
+    if column_exists(cr, "ir_filters", "sort"):
+        cr.execute(
+            """
+               WITH to_update AS (
+                   SELECT id,
+                          COALESCE(array_to_json(array_agg(sort_item ORDER BY rn) filter (WHERE sort_item not in %s)), '[]') AS sort
+                   FROM (
+                       SELECT f.id, s.*
+                         FROM ir_filters f
+                 JOIN LATERAL jsonb_array_elements_text(f.sort::jsonb)
+                       WITH ORDINALITY AS s(sort_item, rn)
+                          ON true
+                       WHERE f.model_id = %s
+                         AND f.sort ~ %s
+                   ) AS matches
```

Comment:
I simplified the query afterward. See last version https://github.com/odoo/upgrade-util/pull/52#discussion_r1516498360

Reply:
Do we need the sorting complexity? Why not something like below?
```sql
SELECT f.id,
        min(f.sort),
        array_to_json(array_remove(array_remove(array_agg(e.value), 'login'), 'login desc'))::text
   FROM ir_filters f
   JOIN LATERAL json_array_elements_text(f.sort::JSON) e
     ON TRUE
  WHERE f.model_id='res.users'
    AND f.sort ~ '\ylogin\y'
  GROUP BY f.id
```
This avoids ordering and such. The `min(f.sort)` was just for quick test
```
+----+------------------------------+-----------------+
| id | min                          | array_to_json   |
|----+------------------------------+-----------------|
| 1  | ["login","lang","name"]      | ["lang","name"] |
| 3  | ["login desc","lang","name"] | ["lang","name"] |
| 2  | ["login","lang","name"]      | ["lang","name"] |
+----+------------------------------+-----------------
```

```

Reply:
- `array_agg` will use an arbitrary order if none specified.
- `array_remove` can be used but `filter` have the same effect.